bazaarvoice/emodb
# Added Limit checks for subscription ttl and event ttl. Goin…

…g forward users will not be able to create subscriptions with high number ttls.

## Github Issue #

We had a RED incident in the past where there was a live subscription with eventTTL of 7,000 days. The subscriber basically took a 1 week TTL, presumably from some sample code, and multiplied it by 1,000.  This lead to a subscription which, on 19-NOV-2018, passed a threshold such that adding that long TTL to the current time led to an integer overflow in the Cassandra thrift implementation, thus raising exceptions and causing fanout to fail.

This PR simply adds a check at the Resource level to avoid creating bad subscriptions with such high Event TTL numbers. 
Also, took this opportunity to add a check for a subscription TTL max of 3 years. I believe that should be fine from users perspective as 3 years is a big number.   

Added tests to verify the added checks. 

## How to Test and Verify

Run the newly added tests. 
Verify that there are no regressions by running full test suite.

## Risk
* Low.
